Lib/multiprocessing/semaphore_tracker.py

```diff
@@ -68,7 +76,17 @@ def ensure_running(self):
                 exe = spawn.get_executable()
                 args = [exe] + util._args_from_interpreter_flags()
                 args += ['-c', cmd % r]
+                # Register a signal mask that will block the signals. This signal
+                # mask will be inherited by the child that is going to be spawned
+                # and will protect the child from a race condition that can make
+                # the child die before it registers signal handlers for SIGINT and
+                # SIGTERM. The mask is unregistered after spawning the child.
+                # For more info, see: https://bugs.python.org/issue33613
+                if _HAVE_SIGMASK:
+                    signal.pthread_sigmask(signal.SIG_BLOCK, _IGNORED_SIGNALS)
```

Comment:
I'm a bit uneasy with this. What if `spawnv_passfds` takes a very long time for some reason? The user will try to stop it using ^C... and nothing will happen.

Reply:
The best would be to use posix_spawn() which supports to set the signal mask when spawning a new child process. But posix_spawn() API is still changing (PR #6693) and it's not available on all platforms. Maybe we can implement the sigmask inside spawnv_passfds(), to prepare the code to be able to use posix_spawn() later?